typing/mode.ml

```diff
@@ -2322,10 +2305,13 @@ module Modality = struct
       | Const c -> c
       | Undefined -> Misc.fatal_error "modality Undefined should not be zapped."
       | Diff (mm, m) ->
-        let c = Mode.zap_to_floor m in
-        let m = Const.Join_const c in
-        (* To give the best modality, we try to cross modality. *)
-        Const.modality_cross_left ~mm m
+        let m = Mode.zap_to_floor m in
+        (* For soundness, we want some [c] such that [m <= join c mm], which
```

Comment:
This is all the dual of what was in the previous commit. Why does this change not change any tests?

Reply:
I'm not sure what you mean - this change is in its own commit, which is just a refactoring.

Reply:
Oh yes I see now.